glusterd2/xlator/options/options.go

```diff
@@ -70,6 +82,164 @@ type Option struct {
 // Option.
 // Returns are error if it is not possible, nil otherwise.
 func (o *Option) Validate(val string) error {
-	// TODO: Do actual validation
+	var err error
+	switch t := o.Type; t {
+	case OptionTypeBool:
+		err = ValidateBool(o, val)
+	case OptionTypeClientAuthAddr:
+		err = ValidateClientAuthAddr(o, val)
+	case OptionTypeDouble:
+		err = ValidateDouble(o, val)
+	case OptionTypeInt:
+		err = ValidateInt(o, val)
+	case OptionTypeInternetAddress:
+		err = ValidateInternetAddress(o, val)
+	case OptionTypeInternetAddressList:
+		err = ValidateInternetAddressList(o, val)
+	case OptionTypePath:
+		err = ValidatePath(o, val)
+	case OptionTypePercent:
+		err = ValidatePercent(o, val)
+	case OptionTypePercentOrSizet:
+		err = ValidatePercentOrSize(o, val)
+	case OptionTypePriorityList:
+		err = ValidatePriorityOrSize(o, val)
+	case OptionTypeSizeList:
+		err = ValidateSizeList(o, val)
+	case OptionTypeSizet:
+		err = ValidateSizet(o, val)
+	case OptionTypeStr:
+		err = ValidateStr(o, val)
+	case OptionTypeTime:
+		err = ValidateTime(o, val)
+	case OptionTypeXlator:
+		err = ValidateXlator(o, val)
+	default:
+		err = ValidateOption(o, val)
+	}
+	return err
+}
+
+// ValidateBool validates if the option is of type boolean
+func ValidateBool(o *Option, val string) error {
+	if val == "" {
+		err := errors.New("No argument passed")
+		return err
+	}
+	switch strings.ToLower(val) {
+	case "on", "yes", "true", "enable", "0", "off", "no", "false", "disable", "1":
+		return nil
+	default:
+		return ErrInvalidArg
+	}
+}
+
+// ValidateClientAuthAddr validates mount auth address
+func ValidateClientAuthAddr(o *Option, val string) error {
+	return ErrInvalidArg
+}
+
+//ValidateDouble validates if the option is of type double
+func ValidateDouble(o *Option, val string) error {
+	if validate.IsFloat(val) != true {
+		return ErrInvalidArg
+	}
 	return nil
 }
+
+// ValidateInt validates if the option is of type Int
+func ValidateInt(o *Option, val string) error {
+	if validate.IsInt(val) != true {
+		return ErrInvalidArg
+	}
```

Comment:
Good. Along with validating if the value can be parsed as the correct type, you will also need to validate if it is in the right range.

If either `o.Min` or `o.Max` are not zero, it should be ensured that the value is in the range. Depending on `o.ValidateType` you will need to check both min and max (default) or only min or only max.

The same needs to be done for Double and Size types as well. Have a common function to validate range.

Reply:
Added ValidateRange() function to check if option in expected range or not.